primitives/arithmetic/src/lib.rs

```diff
@@ -226,7 +226,7 @@ where
 		// we move to the next minimum.
 		let mut min_index = 0;
 		// at this threshold we move to next index.
-		let threshold = targeted_sum / count_t;
+		let threshold = targeted_sum.checked_div(&count_t).unwrap();
```

Comment:
NO UNWRAPS!!

Either provide a strict proof and place it in a `.expect(...)`, or if it can necessarily error due to external factors (e.g. the arguments), then return an `Err(...)`.

Reply:
that is good to know, I will roll it back and keep it in mind when it appears again while working on the changes